File: C++/DSA/Problems/Leetcode/LC1480.cpp
#include <iostream>
#include <vector>
using namespace std;

vector<int> runningSum(vector<int>& nums) {
    for(int i = 1; i < nums.size(); i++){
        nums[i] = nums[i - 1] + nums[i];
    }
    return nums;
}

int main() {
    vector<int> nums = {1, 2, 3, 4, 5};
    vector<int> result = runningSum(nums);

    for (int num : result) {
        cout << num << " ";
    }
    cout << endl;

    return 0;
}

File: Python/DSA/Problems/Leetcode/LC1480.py
from typing import List

class Solution:
    def runningSum(self, nums: List[int]) -> List[int]:
        runningSum = []
        total = 0
        for num in nums:
            total += num
            runningSum.append(total)
        return runningSum


def main():
    sol = Solution()

    test_cases = [
        [1, 2, 3, 4],
        [1, 1, 1, 1, 1],
        [3, 1, 2, 10, 1],
        [],
        [-1, 5, -3, 4],
    ]

    for nums in test_cases:
        print(f"Input:  {nums}")
        print(f"Output: {sol.runningSum(nums)}")
        print("-" * 30)


if __name__ == "__main__":
    main()
